_posts/2022-06-02-building-a-modern-data-stack-for-qaz-ai.md

```diff
@@ -0,0 +1,368 @@
+factor 3000.
+
+
+<p align="center">
+   <img align="center" width="50%" src="assets/blog/qaz-ai-modern-data-stack/tpch-schema.png"/>
+</p>
+
+
+
+For testing, we will consider three tables: _lineitem_ (18 billion rows),
+_orders_ (450 million rows) and _partsupp_ (2.4 billion rows). 
+
+```sql
+create table iceberg2.ice.tpch_sf3000_customer –(450 M)
+with (format = 'ORC')
+as
+select *
+from tpch.sf3000.customer
+
+create table iceberg2.ice.tpch_sf3000_lineitem –(18 B)
+with (format = 'ORC')
+as
+select *
+from tpch.sf3000.lineitem
+
+create table iceberg2.ice.tpch_sf3000_partsupp –(2,4 B)
+with (format = 'ORC')
+as
+select *
+from tpch.sf3000.partsupp
+
+```
+
+Then, we try to join all three of these tables as it is shown in the ER diagram.
+Let’s make it more challenging by turning off one of the workers, which should
+result in a query failure. To enable the automatic query rerun of the failed one
+we will set `retry_policy=QUERY` in `config. properties`.
+
+```sql
+create table iceberg2.ice.tpch_sf3000_lineitem_joined 
+with (format = 'ORC')
+as
+select litem.orderkey ,
+	litem.partkey ,
+	litem.suppkey ,
+	litem.linenumber ,
+	litem.quantity ,
+	litem.extendedprice ,
+	litem.discount ,
+	litem.tax ,
+	litem.returnflag ,
+	litem.linestatus ,
+	litem.shipdate ,
+	litem.commitdate ,
+	litem.receiptdate ,
+	litem.shipinstruct ,
+	litem.shipmode ,
+	litem.comment,
+	psupp.availqty ,
+	psupp.supplycost ,
+	ord.shippriority ,
+	ord.totalprice 
+from iceberg2.ice.tpch_sf100000_lineitem litem
+left join iceberg2.ice.tpch_sf100000_partsupp psupp on litem.partkey = psupp.partkey and litem.suppkey = psupp.suppkey 
+left join iceberg2.ice.tpch_sf100000_orders ord on litem.orderkey = ord.orderkey 
+```
+
+The query has been completed in 4 hours. Also, at query processing, worker 22
+has been turned off. The query has been automatically started over and completed
+successfully. At the query processing, three tables have been joined (_the
+triple join_): 18 billion rows x 2,4 billion rows x 450 million rows. 
+
+
+
+<p align="center">
+   <img align="center" width="50%" src="assets/blog/qaz-ai-modern-data-stack/trino-query-details.png"/>
+</p>
+
+
+
+This experiment gave us the confidence to move forward in our plans to rebuild
+our architecture with Trino in order to perform analytical and transformational
+manipulations upon data directly in S3, which will allow us to exclude HDFS and
+Hive interference in these processes. 
+
+
+<p align="center">
+   <img align="center" width="50%" src="assets/blog/qaz-ai-modern-data-stack/new-architecture.png"/>
+</p>
+
+
+
+As a result we will achieve faster pipelines. 
+
+A huge thanks to the Trino development team and the Trino community for an
+excellent product, which I enjoy using and allows me to go beyond conventional
+usage patterns.
+
+If you are looking for help building your data warehouse, or if you’re
+interested in joining us at QazAI, feel free to reach out to me at
+kuspayev@aifund.kz
```

Comment:
Not sure we should do that, lets remove the email.. instead maybe mention the username on the trino slack and get users to ask you there .. also maybe refer to the project-tardigrade channel